indexprovider/wrapper.go

```diff
@@ -0,0 +1,142 @@
+package indexprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/boost/storagemarket/types/dealcheckpoints"
+	"github.com/hashicorp/go-multierror"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/boost/storagemarket/types"
+	metadata2 "github.com/filecoin-project/index-provider/metadata"
+	"github.com/filecoin-project/lotus/markets/dagstore"
+	"github.com/filecoin-project/lotus/markets/idxprov"
+
+	"github.com/filecoin-project/boost/db"
+	lotus_storagemarket "github.com/filecoin-project/go-fil-markets/storagemarket"
+	provider "github.com/filecoin-project/index-provider"
+	"github.com/ipfs/go-cid"
+)
+
+var log = logging.Logger("index-provider-wrapper")
+
+type Wrapper struct {
+	dealsDB     *db.DealsDB
+	legacyProv  lotus_storagemarket.StorageProvider
+	prov        provider.Interface
+	dagStore    *dagstore.Wrapper
+	meshCreator idxprov.MeshCreator
+}
+
+func NewWrapper(dealsDB *db.DealsDB, legacyProv lotus_storagemarket.StorageProvider, prov provider.Interface, dagStore *dagstore.Wrapper,
+	meshCreator idxprov.MeshCreator) *Wrapper {
+	return &Wrapper{
+		dealsDB:     dealsDB,
+		legacyProv:  legacyProv,
+		prov:        prov,
+		dagStore:    dagStore,
+		meshCreator: meshCreator,
+	}
+}
+
+func (w *Wrapper) IndexerAnnounceAllDeals(ctx context.Context) error {
+	log.Info("will announce all Markets deals to Indexer")
+	err := w.legacyProv.AnnounceAllDealsToIndexer(ctx)
```

Comment:
We don't seem to be doing anything with that error, maybe check it and have a WARN log message if it is != nil ?